solidity/contracts/AugurConstantProductMarketFactory.sol

```diff
@@ -10,6 +10,9 @@ contract AugurConstantProductMarketFactory {
     using ContractExists for address;
 	using AddressToString for address;
 
+	mapping(address => bool) private markets;
+	address[] private marketList;
```

Comment:
Seems useful to be able to index directly into these without needing a helper.  This would also allow us to delete `getIsValidMarket` function.  Could we also get rid of `getNumMarkets`?  Is the length of a public array also public?

```suggestion
	mapping(address => bool) public markets;
	address[] public marketList;
```

Reply:
ah sure. Will tests length too